### PR TITLE
Make /login show descriptive error when an incorrect password is submitted

### LINKED ISSUE #2640: [Bug]: Generic error on incorrect password submission
### Verified issue does not already exist?

- [X] I have searched and found no existing issue
- [X] I will be providing steps how to reproduce the bug (in most cases this will also mean uploading a demo budget file)

### What happened?

When submitting a blank or incorrect password, a generic error message is shown as opposed to "Invalid password".

To reproduce, submit an incorrect password to the actual login page `/login`
image

### Where are you hosting Actual?

Locally via Yarn

### What browsers are you seeing the problem on?

Chrome

### Operating System

Windows 11

### PR DESCRIPTION
**Pull request by matt-fidd**
Fixes #2640 

Before:
image

After:
image

### CODE REVIEW

File: packages/loot-core/src/server/main.ts [hunk 1]
```diff
@@ -1482,16 +1482,18 @@ handlers['subscribe-change-password'] = async function ({ password }) {
...
-  if (res.token) {
-    await asyncStorage.setItem('user-token', res.token);
-    return {};
+    if (res.token) {
+      await asyncStorage.setItem('user-token', res.token);
+      return {};
+    }
+  } catch (e) {
+    return { error: 'invalid-password' };
   }
```
Comment (MatissJanis) [inline packages/loot-core/src/server/main.ts:1496]: **💬 suggestion:** ‏we are missing a return statement if `res.token` is not set or empty. While _theoretically_ it should never happen - we should still account the off chance that it does happen. Maybe throwing a new error would be sufficient in this case.

Comment (matt-fidd) [reply packages/loot-core/src/server/main.ts:1498]: Good idea, added now. Looking at the rest of the file I'm not sure what the convention with thrown error messages is so if you'd prefer it to read differently please let me know!

**Comment by matt-fidd** [comment]
I'm not sure what happened with the Electron tests here, is someone able to rerun them please?

**Review by MatissJanis — APPROVED** [review]
Thanks!

**Pull request merged**